cli/lib/kontena/stacks_cache.rb

```diff
@@ -62,39 +57,39 @@ class RegistryClientFactory
     end
 
     class << self
-      def pull(stack, version = nil)
-        cache(stack, version).read
+      def pull(stack_name)
+        cache(stack_name).read
       end
 
       def dputs(msg)
         Kontena.logger.debug { msg }
       end
 
-      def cache(stack, version = nil)
-        stack = CachedStack.new(stack, version)
+      def cache(stack_name)
+        stack = CachedStack.new(stack_name)
         if stack.cached?
           dputs "Reading from cache: #{stack.path}"
         else
-          dputs "Retrieving #{stack.stack}:#{stack.version} from registry"
-          content = client.pull(stack.stack, stack.version)
-          yaml    = ::YAML.safe_load(content, [], [], true, stack.stack)
-          new_stack = CachedStack.new(yaml['stack'], yaml['version'])
+          dputs "Retrieving #{stack.stack_name} from registry"
+          content = client.pull(stack_name)
+          yaml    = ::YAML.safe_load(content, [], [], true, stack.stack_name.to_s)
+          new_stack_name = Kontena::Cli::Stacks::StackName.new(yaml['stack'])
+          new_stack_name.version = yaml['version']
```

Comment:
I don't quite understand reasons behind this. If `StackName` needs `version` then why it's not passed in `.new`?

Reply:
Then this would have to be:

```
new_stack_name = Kontena::Cli::Stacks::StackName.new([yaml['stack'], yaml['version']].compact.join(':'))
```

or the `StackName#initialize` would need to accept an optional version argument.

Reply:
Hmm actually it does.